common/services/reference-data.js

```diff
@@ -49,14 +49,23 @@ function getLocationById(id) {
   return apiClient.find('location', id).then(response => response.data)
 }
 
-function getLocationsById(locations = []) {
-  const locationPromises = locations.map(locationId => {
-    return getLocationById(locationId).catch(() => false)
-  })
+function getLocationByAgencyId(agencyId) {
+  return getLocations({ nomisAgencyId: agencyId }).then(results => results[0])
+}
+
+function getLocationsById(ids = []) {
+  return _mapLocationIdsToLocations(ids, getLocationById)
+}
 
-  return Promise.all(locationPromises).then(locations =>
-    locations.filter(Boolean)
-  )
+function getLocationsByAgencyId(ids = []) {
+  return _mapLocationIdsToLocations(ids, getLocationByAgencyId)
+}
+
+function _mapLocationIdsToLocations(ids, callback) {
+  const locationPromises = ids.map(id => {
+    return callback(id).catch(() => false)
+  })
+  return Promise.all(locationPromises).then(result => result.filter(Boolean))
```

Comment:
Maybe leave `result` as `locations`. Feels more clear to what is returned from those promises.